src/ex1/main.js

```diff
@@ -0,0 +1,233 @@
+    pendingToDo.textContent = this.listOfToDoArr.length;
+
+    if (!this.listOfToDoArr.length) {
+      deleteAllBtn.classList.remove("active");
+    } else {
+      deleteAllBtn.classList.add("active");
+    }
+    let newToDoTag = "";
+    this.listOfToDoArr.forEach((toDo, index) => {
+      newToDoTag += `<li>${toDo}<span class="icon" onclick="main.deleteTask(${index})"><i class="fas fa-trash"></i></span></li>`;
+    });
+    todoList.innerHTML = newToDoTag;
+    inputText.value = "";
+  }
+}
+
+// Controller - manage the item adding/removing + pokemon fetching
+class ItemManager {
+  //check type of input
+  async addToDo(userEnteredValue) {
+    let listOfPokemons = new Array();
+    let listOfPokemonsToInsert = new Array();
+    let resolvedPromises = new Array();
+    const isPokemonID = this.isInputNumber(userEnteredValue);
+    const isPokemonListID = this.isInputNumberArray(userEnteredValue);
+    // single pokemon id
+    if (isPokemonID) {
+      const pokemonName = await this.fetchFromApi(Number(userEnteredValue));
+      userEnteredValue =
+        typeof pokemonName !== "undefined"
+          ? `Catch ${pokemonName}`
+          : `Oops! Such pokemon was not found yet`;
+    }
+    // List of pokemon ID
+    else if (isPokemonListID.length) {
+      listOfPokemons = await Promise.all([
+        resolvedPromises = await Promise.all(isPokemonListID.map(pokemonID => this.fetchFromApi(pokemonID)))
+      ]).catch((err) => {
+        console.log(err);
+      });
+      resolvedPromises.map((pokemon) => {
+        userEnteredValue =
+          typeof pokemon !== "undefined"
+            ? `Catch ${pokemon}`
+            : `Oops! Such pokemon was not found yet`;
+        listOfPokemonsToInsert.push(userEnteredValue);
+      });
+    }
+
+    const getLocalStorageData = localStorage.getItem("New Todo");
+    if (getLocalStorageData == null) {
+      main.listOfToDoArr = [];
+    } else {
+      main.listOfToDoArr = JSON.parse(getLocalStorageData);
+    }
+    // If there is a list of pokemons, insert one by one
+    // else insert single value
+    if (listOfPokemonsToInsert.length) {
+      listOfPokemonsToInsert.map((pokemonToInsert) => {
+        main.listOfToDoArr.push(pokemonToInsert);
+      });
+    } else {
+      main.listOfToDoArr.push(userEnteredValue);
+    }
+    localStorage.setItem("New Todo", JSON.stringify(main.listOfToDoArr));
+    main.render();
+  }
+
+  async fetchFromApi(pokemonId) {
+    const pokemon = new PokemonClient(pokemonId);
+    const pokemonName = await pokemon.getPokemon();
+
+    return typeof pokemonName !== "undefined" ? pokemonName : undefined;
+  }
+
+  isInputNumber(inputNumber) {
+    return Number.isInteger(Number(inputNumber)) ? true : false;
+  }
+
+  // if array of numbers, return an array of numbers, else return empty array and the todo will be regular
+  isInputNumberArray(inputArray) {
+    const tempList = inputArray.split(",");
```

Comment:
you could remove spaces (e.g. 1, 2 is not working for the second one - with space before it)